hail_search/hail_search_query.py

```diff
@@ -625,6 +654,63 @@ def _format_transcript_args(self):
         })
         return args
 
+    def _annotate_allowed_consequences(self, annotations, annotation_filters):
+        allowed_consequences = {
+            ann for field, anns in (annotations or {}).items()
+            if anns and (field not in ANNOTATION_OVERRIDE_FIELDS) for ann in anns
+        }
+        consequence_enum = self._get_enum_lookup('sorted_transcript_consequences', 'consequence_term')
+        allowed_consequence_ids = {consequence_enum[c] for c in allowed_consequences if consequence_enum.get(c)}
+
+        annotation_exprs = {}
+        if allowed_consequence_ids:
+            allowed_consequence_ids = hl.set(allowed_consequence_ids)
+            allowed_transcripts = self._ht.sorted_transcript_consequences.filter(
+                lambda tc: tc.consequence_term_ids.any(allowed_consequence_ids.contains)
+            )
+            annotation_exprs['allowed_transcripts'] = allowed_transcripts
+            annotation_filters = annotation_filters + [hl.is_defined(allowed_transcripts.first())]
+
+        consequence_filter = self._or_filter(annotation_filters)
+        if consequence_filter is not None:
+            annotation_exprs['has_allowed_annotation'] = consequence_filter
+
+        if annotation_exprs:
+            self._ht = self._ht.annotate(**annotation_exprs)
+
+    def _get_annotation_override_filters(self, pathogenicity, annotations):
```

Comment:
I'm curious about the `override` in the function's name. The function gets annotation filters other than the allowed transcripts. Why is it a kind of `overriding`?

Reply:
These are the filters that override the annotation filter, thus the name `annotation_override_filters`. Certain filters  such as pathogenicity and splice AI override the annotation filter, so the variant is returned even if it does not pass the annotation filter